pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -70,18 +70,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, o results.Object) error {
 		logger.Debugf("Post SetGroupVersionKind: %s", o.GetObjectKind().GroupVersionKind().String())
 	}
 
-	// Update record.
+	// Upsert record.
+	startTime := time.Now()
 	result, record, err := r.resultsClient.Put(ctx, o)
+	timeTakenField := zap.Int64("results.tekton.dev/time-taken-ms", time.Since(startTime).Milliseconds())
```

Comment:
Potential feature - try and capture this in Prometheus metrics. We should discuss what sort of metrics make sense and what level of detail we want to record data.

Reply:
I totally agree. Exposing relevant metrics from both API server and Watcher is an important step towards making the project production ready.